tests/connectors/rds/test_base.py

```diff
@@ -0,0 +1,158 @@
+import pytest
+from unittest.mock import patch
+from pandas import DataFrame, to_datetime
+
+from numalogic.connectors._config import Pivot
+from numalogic.connectors.rds._base import RDSDataFetcher, RDSConfig, Boto3ClientManager
+
+
+class TestRDSDataFetcher:
+    @pytest.fixture(autouse=True)
+    def setUp(self, mocker):
+        self.mock_boto3 = mocker.MagicMock(spec=Boto3ClientManager)
+        self.mock_rds_config = mocker.MagicMock(spec=RDSConfig)
+        self.mock_rds_config.database_type = "db_type"
+        self.data_fetcher = RDSDataFetcher(db_config=self.mock_rds_config)
+
```

Comment:
This is still in unittest format. For example using pytest, check
https://github.com/numaproj/numalogic/blob/main/tests/udfs/test_inference.py

Reply:
I did not get it, which actually you are referring as unittest format ?

Reply:
make it function based rather than class based

Reply:
and let's use fixtures